language/bytecode_verifier/src/type_memory_safety.rs

```diff
@@ -791,14 +869,17 @@ impl<'a> TypeAndMemorySafetyAnalysis<'a> {
                     return Err(VMStaticViolation::GlobalReferenceError(offset));
                 }
 
+                let tys = &self.module().locals_signature_at(*tys_idx).0;
+                let struct_type =
+                    SignatureToken::Struct(struct_definition.struct_handle, tys.clone());
+                SignatureTokenView::new(self.module(), &struct_type).kind(self.type_formals())?;
```

Comment:
The result of this line does not seem to be captured in any variable.

Reply:
`fn kind` also verifies the kind correctness of the type recursively. Would not capturing the result with a variable causing it to be optimized out by the compiler?

Reply:
ok, I guess this comment is related to the other comment about refactoring the code so that the code in views.rs provides only a view and the actual checking is done in bytecode verifier.

Reply:
correct.